### PR TITLE
Added samples for Section, Content-Control and Footnotes and Endnotes

### PR DESCRIPTION
**Pull request by ManikandanRavichandranSync**
Added samples for Section, Content-Control and Footnotes and Endnotes

### CODE REVIEW

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text/Program.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                        break;
+                }
+            }
+        }
+        /// <summary>
+        /// Replace Text With Inline content control
+        /// </summary>
+        /// <param name="text"></param>
+        /// <param name="inlineContentControl"></param>
+        private static void ReplaceTextWithInlineContentControl(string text, InlineContentControl inlineContentControl)
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text/Program.cs]: Addressed the feedback

File: Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                //A Text body has 3 types of elements - Paragraph, Table and Block Content Control
+                //Decides the element type by using EntityType
+                switch (bodyItemEntity.EntityType)
+                {
+                    case EntityType.Paragraph:
+                        WParagraph paragraph = bodyItemEntity as WParagraph;
+                        for (int j = 0; j < paragraph.ChildEntities.Count; j++)
+                        {
+                            if (paragraph.ChildEntities[j] is WFootnote)
+                            {
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs]: Addressed the feedback

File: Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text/Program.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                        IterateTextBody(textBox.TextBoxBody);
+                        break;
+                    case EntityType.Shape:
+                        //Iterates to the body items of shape.
+                        Shape shape = entity as Shape;
+                        IterateTextBody(shape.TextBody);
+                        break;
+                    case EntityType.InlineContentControl:
+                        InlineContentControl inlineContentControl = entity as InlineContentControl;
+                        if(inlineContentControl.ContentControlProperties.Title == "ReplaceText")
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text/Program.cs]: Addressed the feedback

File: Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text.sln [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.28307.136
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Edit-content-control-text", "Edit-content-control-text\Edit-content-control-text.csproj", "{AE9996D9-17F0-4A69-AB1D-CA10E8605CF7}"
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Content-Controls/Edit-content-control-text/.NET-Standard/Edit-content-control-text.sln]: Addressed the feedback

File: Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                    inputStream.Dispose();
+                    WordDocument newDocument = null;
+                    WSection newSection = null;
+                    int i = 0;
+                    //Iterate each section from Word document
+                    foreach (WSection section in document.Sections)
+                    {
+                        foreach (TextBodyItem textbodyitem in section.Body.ChildEntities)
+                        {
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs:27]: Addressed the feedback

File: Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                            if (textbodyitem is WParagraph)
+                            {
+                                WParagraph para = textbodyitem as WParagraph;
+                                if (para.StyleName == "Heading 1")
+                                {
+                                    if (newDocument != null)
+                                    {
+                                        //Saves the Word document to  MemoryStream
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs]: Addressed the feedback

File: Word-document/Split-by-section/.NET-Standard/Split-by-section/Program.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                {
+                    int i = 0;
+                    //Iterate each section from Word document
+                    foreach (WSection section in document.Sections)
+                    {
+                        //Create new Word document
+                        using (WordDocument newDocument = new WordDocument())
+                        {
+                            //Add cloned section into new Word document
+                            newDocument.Sections.Add(section.Clone());
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Word-document/Split-by-section/.NET-Standard/Split-by-section/Program.cs:25]: Addressed the feedback

File: Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                                        }
+                                        i++;
+                                    }
+                                    //Create new Word document
+                                    newDocument = new WordDocument();
+                                    newSection = newDocument.AddSection() as WSection;
+                                    //Add cloned paragraphs into new section
+                                    newSection.Body.ChildEntities.Add(para.Clone());
+                                }
+                                else if (para.StyleName != "Heading 1" && newDocument != null)
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs]: Addressed the feedback

File: Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes.sln [hunk 9]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.28307.136
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Edit-footnotes-and-endnotes", "Edit-footnotes-and-endnotes\Edit-footnotes-and-endnotes.csproj", "{8EC56111-D683-4E69-8559-4142B0199648}"
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes.sln]: Addressed the feedback

File: Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        }
+        /// <summary>
+        /// Remove FootNote from Word document
+        /// </summary>
+        /// <param name="document"></param>
+        private static void RemoveFootNote(WordDocument document)
+        {
+            foreach (WSection section in document.Sections)
+            {
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs]: Addressed the feedback

File: Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+                        //Saves the Word document to file stream.
+                        document.Save(outputStream, FormatType.Docx);
+                    }
+                }
+            }
+        }
+        /// <summary>
+        /// Remove FootNote from Word document
+        /// </summary>
+        /// <param name="document"></param>
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs]: Addressed the feedback

File: Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                IEntity bodyItemEntity = textBody.ChildEntities[i];
+                //A Text body has 3 types of elements - Paragraph, Table and Block Content Control
+                //Decides the element type by using EntityType
+                switch (bodyItemEntity.EntityType)
+                {
+                    case EntityType.Paragraph:
+                        WParagraph paragraph = bodyItemEntity as WParagraph;
+                        for (int j = 0; j < paragraph.ChildEntities.Count; j++)
+                        {
+                            if (paragraph.ChildEntities[j] is WFootnote)
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Footnotes-and-Endnotes/Edit-footnotes-and-endnotes/.NET-Standard/Edit-footnotes-and-endnotes/Program.cs]: Addressed the feedback

File: Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using (FileStream inputStream = new FileStream(@"../../../Template.docx", FileMode.Open, FileAccess.Read))
+            {
+                //Load the template document as stream
+                using (WordDocument document = new WordDocument(inputStream, FormatType.Docx))
+                {
+                    inputStream.Dispose();
```
[the remark this answers is not in the document]
Comment (ManikandanRavichandranSync) [reply Word-document/Split-by-heading/.NET-Standard/Split-by-heading/Program.cs]: Addressed the feedback

**Review by sivasubramanv — APPROVED** [review]

**Review by sivasubramanv — APPROVED** [review]

**Review by sivasubramanv — APPROVED** [review]

**Comment by sivasubramanv** [comment]
Commit

**Pull request merged**